### PR TITLE
Fix comment dialog dialog bug and refactor PostPage_test

### LINKED ISSUE #706: Fix intermittent JS test timeouts on travis
The tests pass on rerun, but may initially fail due to timeouts, e.g.:

```
  459 passing (1m)
  2 failing
  1) PostPage
       should switch the sorting method when an option is selected:
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at <anonymous>
  2) PostPage
       as a moderator user
         should report a comment:
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at <anonymous>
```

### LINKED ISSUE #1765: Dialog box stays up after deleting a comment
To reproduce, click the comment menu of a comment and click 'Delete'. A toast message says the comment was deleted successfully, and if you refresh you will not see the comment anymore. However the dialog box is not dismissed.

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Fixes #1765 
Fixes #706 

#### What's this PR do?
Fixes the dialog not closing bug and also refactors tests on PostPage. There should be new tests for the bug and other related functionality.

#### How should this be manually tested?
Delete a comment. Click the accept button on the confirmation dialog. The comment should be deleted and the dialog should disappear.

### CODE REVIEW

File: static/js/containers/PostPage_test.js [hunk 1]
```diff
@@ -259,69 +277,68 @@ describe("PostPage", function() {
...
     const moreComments = makeMoreComments(post, parent.id)
     const newComment = makeComment(post, parent.id)
     const newComments = [newComment]
 
     helper.getMoreCommentsStub.returns(Promise.resolve(newComments))
-    await listenForActions(
+    await commentTreeProps.loadMoreComments(moreComments)
+    const actionsList = store.getActions()
+    assert.deepEqual(
+      actionsList.slice(actionsList.length - 3).map(action => action.type),
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: I think you can rewrite this with [`R.pluck`](https://ramdajs.com/docs/#pluck) and [`assert.includeMembers`](https://www.chaijs.com/api/assert/#method_includemembers). Something like:

```javascript
assert.includeMembers(
  R.pluck("type")(actionsList),
  [
    actions.morecomments.get.requestType,	        
    actions.morecomments.get.successType,
    REPLACE_MORE_COMMENTS
  ]
)
```

File: static/js/containers/PostPage_test.js [hunk 2]
```diff
@@ -330,456 +347,780 @@ describe("PostPage", function() {
...
-    const {
-      location: { pathname }
-    } = helper.browserHistory
-    assert.equal(pathname, channelURL(channel.name))
+  describe("deleting post", () => {
+    it("opens a confirmation dialog for deleting a post", async () => {
+      const { inner, store } = await render()
+      inner.find("ExpandedPostDisplay").prop("showPostDeleteDialog")()
+      const actionsList = store.getActions()
+      assert.deepEqual(actionsList[actionsList.length - 1], {
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: `R.last` instead of `[actionsList.length - 1]`?

File: static/js/containers/PostPage_test.js [hunk 3]
```diff
@@ -330,456 +347,780 @@ describe("PostPage", function() {
...
+      const props = inner
+        .find("OurDialog")
+        .at(1)
+        .props()
+
+      assert.equal(props.title, "Delete Post")
+      await props.onAccept()
+      sinon.assert.calledWith(helper.deletePostStub, post.id)
+      const actionsList = store.getActions()
+      assert.deepEqual(actionsList.slice(actionsList.length - 4), [
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: `assert.includeMembers` here again if that works

File: static/js/containers/PostPage_test.js [hunk 4]
```diff
@@ -330,456 +347,780 @@ describe("PostPage", function() {
...
+  })
+
+  describe("deleting comment", () => {
+    it("opens a confirmation dialog for deleting a comment", async () => {
+      const { inner, store } = await render()
+      const comment = comments[0]
+
+      inner.find("CommentTree").prop("deleteComment")(comment)
+      const actionsList = store.getActions()
+      assert.deepEqual(actionsList.slice(actionsList.length - 2), [
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: `assert.includeMembers`

File: static/js/containers/PostPage_test.js [hunk 5]
```diff
@@ -330,456 +347,780 @@ describe("PostPage", function() {
...
+      const dialogProps = inner
+        .find("OurDialog")
+        .at(0)
+        .props()
+      assert.equal(dialogProps.title, "Delete Comment")
+      await dialogProps.onAccept()
+
+      const actionsList = store.getActions()
+      sinon.assert.calledWith(helper.deleteCommentStub, comment.id)
+      assert.deepEqual(actionsList.slice(actionsList.length - 5), [
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: Same as above

File: static/js/containers/PostPage_test.js [hunk 6]
```diff
@@ -330,456 +347,780 @@ describe("PostPage", function() {
...
-          wrapper
+    describe("removing or approving", () => {
+      describe("posts", () => {
+        it("should show a dialog to remove the post", async () => {
+          post.removed = false
+          const { inner, store } = await render()
+
+          await inner.find("ExpandedPostDisplay").prop("removePost")(post)
+          const actionsList = store.getActions()
+          assert.deepEqual(actionsList.slice(actionsList.length - 2), [
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: Same as above

File: static/js/containers/PostPage_test.js [hunk 7]
```diff
@@ -330,456 +347,780 @@ describe("PostPage", function() {
...
             .at(1)
             .props()
-            .onAccept({ preventDefault: helper.sandbox.stub() })
-        }
-      )
+          assert.equal(dialogProps.title, "Remove Post")
+          await dialogProps.onAccept({ preventDefault: helper.sandbox.stub() })
+
+          const actionsList = store.getActions()
+          assert.deepEqual(actionsList.slice(actionsList.length - 6), [
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: Same as above

File: static/js/containers/PostPage_test.js [hunk 8]
```diff
@@ -330,456 +347,780 @@ describe("PostPage", function() {
...
+          const expected = {
+            ...post,
+            removed: false
+          }
+          helper.updateRemovedStub.returns(Promise.resolve(expected))
+
+          await inner.find("ExpandedPostDisplay").prop("approvePost")(post)
+          const actionsList = store.getActions()
+
+          assert.deepEqual(actionsList.slice(actionsList.length - 1), [
```
Comment (gsidebo) [inline static/js/containers/PostPage_test.js]: Same as above

**Comment by noisecapella** [comment]
I replaced all the `.slice` usages in the file. Can you take a second look?

**Pull request merged**